src/robusta/integrations/jira/client.py

```diff
@@ -131,6 +155,122 @@ def _get_default_project_id(self):
+                    {
+                        "content": [
+                            {
+                                "text": "Issue was marked as resolved by the Alertmanager",
+                                "type": "text",
+                            }
+                        ],
+                        "type": "paragraph",
+                    }
+                ],
+                "type": "doc",
+                "version": 1,
+            }
+        }
+        logging.debug(f"Adding comment to issue {issue_id} with payload: {payload}")
+        response = self._call_jira_api(url, http_method=HttpMethod.POST, json=payload) or {}
+        logging.debug(response)
+
+    def create_issue(self, issue_data, issue_attachments=None):
+        endpoint = "issue"
+        url = self._get_full_jira_url(endpoint)
+        payload = {
+            "update": {},
+            "fields": {
+                **issue_data,
+                "issuetype": {"id": str(self.default_issue_type_id)},
+                "project": {"id": str(self.default_project_id)},
+            },
+        }
+        logging.debug(f"Create issue with payload: {payload}")
+        response = self._call_jira_api(url, http_method=HttpMethod.POST, json=payload) or {}
+        logging.debug(response)
+        issue_id = response.get("id")
+        if issue_id and issue_attachments:
+            self.add_attachment(issue_id, issue_attachments)
+
+    def update_issue(self, issue, issue_data):
+        issue_id = self._get_dict_field(issue, "id", -1)
+        summary = self._get_dict_field(issue_data, "summary", "")
+        description = self._get_dict_field(issue_data, "description", "")
+        endpoint = f"issue/{issue_id}"
+        url = self._get_full_jira_url(endpoint)
+
+        payload = {
+            "fields": {"summary": summary, "description": description},
+        }
+
+        logging.debug(f"Update issue '{issue_id}' with payload: {payload}")
+        response = self._call_jira_api(url, http_method=HttpMethod.PUT, json=payload) or {}
+        logging.debug(response)
+
+    def manage_issue(self, issue_data, alert_data, issue_attachments=None):
+        existing_issue = self._get_issue_for_labels(issue_data.get("labels", [])) or []
+        status = self._get_dict_field(alert_data, "status", -1)
+        source = self._get_dict_field(alert_data, "source", None)
+        alert_resolved = status == FindingStatus.RESOLVED
+        is_prometheus_alert = source == FindingSource.PROMETHEUS
+
+        if existing_issue and is_prometheus_alert:
+            issue_done = self._check_issue_done(existing_issue)
+
+            if issue_done:
+                if alert_resolved:
+                    logging.warn("Ignoring resolved alert that is already 'done' in Jira")
+                elif self.reopenIssues:
+                    self.transition_issue(existing_issue, self.reopenStatusName)
+                    self.update_issue(existing_issue, issue_data)
```

Comment:
Should we add more data when reopening the issue?
Maybe just a comment that the issue was reopened at this time?